src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -50,42 +52,54 @@ class StreamReader {
     [[nodiscard]] auto get_num_events_buffered() const -> size_t;
 
     /**
-     * Deserializes and buffers log events in the range `[beginIdx, endIdx)`. After the stream has
-     * been exhausted, it will be deallocated.
+     * @return The filtered log events map.
+     */
+    [[nodiscard]] auto get_filtered_log_event_map() const -> FilteredLogEventMapType;
+
+    /**
+     * Generates a filtered collection from all log events.
      *
-     * NOTE: Currently, this class only supports deserializing the full range of log events in the
-     * stream.
+     * @param logLevelFilter Array of selected log levels
+     */
+    void filter_log_events(emscripten::val const& log_level_filter);
+
+    /**
+     * Deserializes all log events in the file. After the stream has been exhausted, it will be
+     * deallocated.
      *
-     * @param begin_idx
-     * @param end_idx
      * @return The number of successfully deserialized ("valid") log events.
      */
-    [[nodiscard]] auto deserialize_range(size_t begin_idx, size_t end_idx) -> size_t;
+    [[nodiscard]] auto build() -> size_t;
```

Comment:
How about `deserialize_stream`? From a library perspective (i.e., ignoring how the log viewer uses this library), `build` sounds like it's going to build the `StreamReader`, not deserialize all events in the stream.

Reply:
Sure, but this library needs to start somewhere and we don't have time to implement *all* of the IR methods right now, right? It would also waste more time to create a log-viewer-specific accelerator library just to eventually throw it away and replace it with this library. Fwiw, `build` is a pretty poor name in general, even in the log viewer, lol.